controllers/grid/navigationMenus/form/PKPNavigationMenuItemsForm.inc.php

```diff
@@ -95,38 +95,53 @@ function fetch($request, $template = 'controllers/grid/navigationMenus/form/navi
 			}
 		}
 
-		$templateMgr->assign(array(
+		$templateArray = array(
 			'navigationMenuItemTypeTitles' => $typeTitles,
 			'navigationMenuItemTypeDescriptions' => json_encode($typeDescriptions),
 			'navigationMenuItemTypeConditionalWarnings' => json_encode($typeConditionalWarnings),
-		));
+		);
+
+		$templateMgr->assign($templateArray);
 
 		return parent::fetch($request, $template, $display);
 	}
 
 	/**
 	 * Initialize form data from current navigation menu item.
 	 */
-	function initData() {
+	function initData($data = array()) {
```

Comment:
This will cause a PHP Warning in PHP 7+ because `Form::initData()` does not have any arguments. 

I think you can accomplish what you want by using `$this->_data` in the OMP `NavigationMenuItemsForm::initData()`. So instead of calling `parent::initData($yourArray)` at https://github.com/pkp/omp/pull/592/files#diff-f1433db460771e9f6a46a89b7b158967R75, you would assign to `$this->_data`:

```
$this->_data = $yourArray;
parent::initData();
```

And then at line 124 below, you'd do the following:

```
$this->_data = array_merge($this->_data, $formData);
```

Reply:
Or actually  maybe reverse it so that the app-specific data overrides the lib data:

```
$this->_data = array_merge($formData, $this->_data);
```